packages/react-native-executorch/common/rnexecutorch/models/ocr/DetectorUtils.cpp

```diff
@@ -0,0 +1,647 @@
+    rotatedPoints[i] = rotatedPoint;
+  }
+
+  return rotatedPoints;
+}
+
+float calculateMinimalDistanceBetweenBox(const std::array<Point, 4> &box1,
+                                         const std::array<Point, 4> &box2) {
+  float minDistance = std::numeric_limits<float>::max();
+  for (const Point &corner1 : box1) {
+    for (const Point &corner2 : box2) {
+      const float distance = distanceFromPoint(corner1, corner2);
+      if (distance < minDistance) {
+        minDistance = distance;
+      }
+    }
+  }
+  return minDistance;
+}
+
+/**
+ * Orders a set of points in a clockwise direction starting with the top-left
+ * point.
+ *
+ * Process:
+ * 1. It iterates through each Point.
+ * 2. For each point, it calculates the sum (x + y) and difference (y - x) of
+ * the coordinates.
+ * 3. Points are classified into:
+ *    - Top-left: Minimum sum.
+ *    - Bottom-right: Maximum sum.
+ *    - Top-right: Minimum difference.
+ *    - Bottom-left: Maximum difference.
+ * 4. The points are ordered starting from the top-left in a clockwise manner:
+ * top-left, top-right, bottom-right, bottom-left.
+ */
+std::array<Point, 4> orderPointsClockwise(const std::array<Point, 4> &points) {
+  Point topLeft, topRight, bottomRight, bottomLeft;
+  float minSum = std::numeric_limits<float>::max();
+  float maxSum = std::numeric_limits<float>::lowest();
+  float minDiff = std::numeric_limits<float>::max();
+  float maxDiff = std::numeric_limits<float>::lowest();
+
+  for (const Point pt : points) {
+    const float sum = pt.x + pt.y;
+    const float diff = pt.y - pt.x;
+
+    if (sum < minSum) {
+      minSum = sum;
+      topLeft = pt;
+    }
+    if (sum > maxSum) {
+      maxSum = sum;
+      bottomRight = pt;
+    }
+    if (diff < minDiff) {
+      minDiff = diff;
+      topRight = pt;
+    }
+    if (diff > maxDiff) {
+      maxDiff = diff;
+      bottomLeft = pt;
+    }
+  }
+
+  return {topLeft, topRight, bottomRight, bottomLeft};
+}
+
+std::vector<cv::Point2f>
+cvPointsFromPoints(const std::array<Point, 4> &points) {
+  std::vector<cv::Point2f> cvPoints;
+  for (const Point &point : points) {
+    cvPoints.emplace_back(point.x, point.y);
+  }
+  return cvPoints;
+}
+
+std::array<Point, 4> pointsFromCvPoints(cv::Point2f cvPoints[4]) {
+  std::array<Point, 4> points;
+  for (std::size_t i = 0; i < 4; ++i) {
+    points[i] = {.x = cvPoints[i].x, .y = cvPoints[i].y};
+  }
+  return points;
+}
+
+std::array<Point, 4> mergeRotatedBoxes(std::array<Point, 4> &box1,
+                                       std::array<Point, 4> &box2) {
+  box1 = orderPointsClockwise(box1);
+  box2 = orderPointsClockwise(box2);
+
+  std::vector<cv::Point2f> points1 = cvPointsFromPoints(box1);
+  std::vector<cv::Point2f> points2 = cvPointsFromPoints(box2);
+
+  std::vector<cv::Point2f> allPoints;
+  allPoints.insert(allPoints.end(), points1.begin(), points1.end());
+  allPoints.insert(allPoints.end(), points2.begin(), points2.end());
```

Comment:
Do you need box1 and box2 after the insertion? If not, use `std::make_move_iterator`